File: generated_code/phi4/baseline_code_java/code_row_122.java
public class MemoryManagementExample {

    public static void main(String[] args) {
        // Allocate first chunk of memory
        int[] firstChunk = new int[10];
        
        // Initialize the first chunk with some values (for demonstration)
        for (int i = 0; i < 10; i++) {
            firstChunk[i] = i;
        }

        // Allocate second chunk of memory
        int[] secondChunk = new int[10];

        // Copy contents from the first chunk to the second chunk
        System.arraycopy(firstChunk, 0, secondChunk, 0, 10);

        // "Free" the first chunk (in Java, this is done by dereferencing)
        firstChunk = null;

        // Suggest garbage collection for cleanup (though not guaranteed)
        System.gc();

        // Print contents of second chunk to verify copy
        for (int value : secondChunk) {
            System.out.println(value);
        }

        // "Free" the second chunk
        secondChunk = null;

        // Suggest garbage collection again
        System.gc();
    }
}

File: generated_code/phi4/baseline_code_c/code_row_122.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    char *first_chunk = (char *)malloc(10);
    if (!first_chunk) {
        return 1;
    }
    
    char *second_chunk = (char *)malloc(10);
    if (!second_chunk) {
        free(first_chunk);
        return 1;
    }

    // Initialize first chunk with some data
    for (int i = 0; i < 10; ++i) {
        first_chunk[i] = 'A' + i;
    }
    
    // Copy contents of the first chunk into the second chunk
    memcpy(second_chunk, first_chunk, 10);
    
    // Free the first chunk
    free(first_chunk);

    // Display content of the second chunk
    for (int i = 0; i < 10; ++i) {
        printf("%c ", second_chunk[i]);
    }
    printf("\n");

    // Free the second chunk
    free(second_chunk);
    
    return 0;
}